#! /bin/sh /usr/share/dpatch/dpatch-run
## ddns-update-style-default.dpatch by  <apollock@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch to provide a sane upgrade path from DHCP v2, defaulting
## DP: ddns-update-style to none

@DPATCH@
diff -urNad dhcp3-3.0.2/server/dhcpd.c /tmp/dpep.mAiHoU/dhcp3-3.0.2/server/dhcpd.c
--- dhcp3-3.0.2/server/dhcpd.c	2004-09-30 09:01:50.000000000 +1000
+++ /tmp/dpep.mAiHoU/dhcp3-3.0.2/server/dhcpd.c	2005-06-15 12:43:06.782036067 +1000
@@ -753,6 +753,13 @@
 		data_string_forget (&db, MDL);
 	}
 
+	/*
+		On Debian we default to 'ddns-update-style none' so we do not
+		break upgrades from the version 2 DHCP server. If we do not
+		default to something dhcpd will refuse to start in the absence
+		of the ddns-update-style statement - peloy@debian.org
+	*/
+	ddns_update_style = 0;
 	oc = lookup_option (&server_universe, options, SV_DDNS_UPDATE_STYLE);
 	if (oc) {
 		if (evaluate_option_cache (&db, (struct packet *)0,
@@ -767,16 +774,6 @@
 				log_fatal ("invalid dns update type");
 			data_string_forget (&db, MDL);
 		}
-	} else {
-		log_info ("%s", "");
-		log_error ("** You must add a global ddns-update-style %s%s.",
-			   "statement to ", path_dhcpd_conf);
-		log_error ("   To get the same behaviour as in 3.0b2pl11 %s",
-			   "and previous");
-		log_error ("   versions, add a line that says \"%s\"",
-			   "ddns-update-style ad-hoc;");
-		log_fatal ("   Please read the dhcpd.conf manual page %s",
-			   "for more information. **");
 	}
 
 	oc = lookup_option (&server_universe, options, SV_LOG_FACILITY);
